sql/core/src/test/resources/sql-tests/results/pgSQL/date.sql.out

```diff
@@ -520,7 +520,7 @@ select make_date(-44, 3, 15)
 -- !query 48 schema
 struct<make_date(-44, 3, 15):date>
 -- !query 48 output
-0045-03-15
+-0044-03-15
```

Comment:
Interesting, so postgres had a different answer here? also shouldn't the year be negative in this output or do I misunderstand?

Reply:
Postgres's answer is `44 BC` for `-44` https://github.com/postgres/postgres/blob/master/src/test/regress/expected/date.out#L1454-L1457 which is wrong. It must be `45 BC` or `-44` (see https://github.com/apache/spark/pull/25210#discussion_r305609274).

Reply:
OK so its answer of "0045" is really 45 BC? OK.

Reply:
Yes, our current answer is `45` which is `45 BC`... but because we skip the era field in our pattern, it is just `45` or `45 CE` (current era by default). That's wrong as well.

So, correct answer must include either **positive year + era** or **year with sign (-)**. PostgreSQL choose the first approach but it converts year incorrectly. I have proposed in this PR the second approach because it allows to maintain only one pattern (`uuuu` instead of `yyyy`).

The first approach requires dynamically switching to the pattern with era only for negative years. I don't think it is worth, especially when SQL standard tells us that years must be in the range `[0001, 9999]`.